javabasics/src/main/java/javabasics/Library.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * This Java source file was generated by the Gradle 'init' task.
+ */
+package javabasics;
+
+import java.util.Random;
+
+
+public class Library {
+    public boolean someLibraryMethod() {
+        return true;
+    }
+
+    public int[] roll(int n){
+
+        int[] rollResult = new int[n];
+
+        for (int i = 0; i < n; i++) {
+
+            Random generator = new Random();
+            Float randomNumber = generator.nextFloat();
+
+            //This feels like the wrong way to get a number between 1 and 6, but it works, so...
+            int randomDice = Math.round(randomNumber * 5) + 1;
+
+            rollResult[i] = randomDice;
+        }
+        return rollResult;
+    }
+
+    public boolean containsDuplicates(String[] arrayToCheck) {
+        boolean hasDups = false;
+        dupCheck:
+
+            for (int i = 0; i < arrayToCheck.length; i++){
+                for (int j = 0; j < arrayToCheck.length; j++) {
+                    if ((arrayToCheck[i].equals(arrayToCheck[j])) && (i != j)) {
+                        hasDups = true;
+                        break dupCheck;
+                    }
+                }
+            }
+            return hasDups;
+    }
+
+    public int calculateAverage(int[] arrayToAverage){
+
+        if (arrayToAverage.length == 1) {
+            return arrayToAverage.length;
+        }
+
+        int sum = 0;
+
+        for (int number : arrayToAverage) {
+            sum = sum + number;
+        }
+
+        return sum/arrayToAverage.length;
+    }
+
+    public int[] lowestAverageArray(int[][] arrayOfArrays){
+
```

Comment:
Will this work with negative values? Instead of instantiating `int winningAvg = 0;` you can set it to the average of the first array in the input, or you can set it to the lowest int possible in Java like this: `int winningAvg = Integer.MIN_VALUE;`  Both of those are good options unless you know for sure you won't be dealing with negative numbers.